core/src/layers/prometheus.rs

```diff
@@ -0,0 +1,1240 @@
+            scheme: scheme.to_string(),
+        }
+    }
+}
+/// [`PrometheusMetrics`] provide the performance and IO metrics.
+#[derive(Debug)]
+pub struct PrometheusMetrics {
+    // metadata
+    pub requests_total_metadata: GenericCounterVec<AtomicU64>,
+    pub requests_duration_seconds_metadata: HistogramVec,
+
+    // create
+    pub requests_total_create: GenericCounterVec<AtomicU64>,
+    pub requests_duration_seconds_create: HistogramVec,
+
+    /// read
+    pub requests_total_read: GenericCounterVec<AtomicU64>,
+    pub requests_duration_seconds_read: HistogramVec,
+    pub bytes_total_read: HistogramVec,
+
+    // write
+    pub requests_total_write: GenericCounterVec<AtomicU64>,
+    pub requests_duration_seconds_write: HistogramVec,
+    pub bytes_total_write: HistogramVec,
+
+    // stat
+    pub requests_total_stat: GenericCounterVec<AtomicU64>,
+    pub requests_duration_seconds_stat: HistogramVec,
+
+    // delete
+    pub requests_total_delete: GenericCounterVec<AtomicU64>,
+    pub requests_duration_seconds_delete: HistogramVec,
+
+    // list
+    pub requests_total_list: GenericCounterVec<AtomicU64>,
+    pub requests_duration_seconds_list: HistogramVec,
+
+    // scan
+    pub requests_total_scan: GenericCounterVec<AtomicU64>,
+    pub requests_duration_seconds_scan: HistogramVec,
+
+    // presign
+    pub requests_total_presign: GenericCounterVec<AtomicU64>,
+    pub requests_duration_seconds_presign: HistogramVec,
+
+    // batch
+    pub requests_total_batch: GenericCounterVec<AtomicU64>,
+    pub requests_duration_seconds_batch: HistogramVec,
+
+    // blocking create
+    pub requests_total_blocking_create: GenericCounterVec<AtomicU64>,
+    pub requests_duration_seconds_blocking_create: HistogramVec,
+
+    // blocking read
+    pub requests_total_blocking_read: GenericCounterVec<AtomicU64>,
+    pub requests_duration_seconds_blocking_read: HistogramVec,
+    pub bytes_total_blocking_read: HistogramVec,
+
+    // blocking write
+    pub requests_total_blocking_write: GenericCounterVec<AtomicU64>,
+    pub requests_duration_seconds_blocking_write: HistogramVec,
+    pub bytes_total_blocking_write: HistogramVec,
+
+    // blocking stat
+    pub requests_total_blocking_stat: GenericCounterVec<AtomicU64>,
+    pub requests_duration_seconds_blocking_stat: HistogramVec,
+
+    // blocking delete
+    pub requests_total_blocking_delete: GenericCounterVec<AtomicU64>,
+    pub requests_duration_seconds_blocking_delete: HistogramVec,
+
+    // blocking list
+    pub requests_total_blocking_list: GenericCounterVec<AtomicU64>,
+    pub requests_duration_seconds_blocking_list: HistogramVec,
+
+    // blocking scan
+    pub requests_total_blocking_scan: GenericCounterVec<AtomicU64>,
+    pub requests_duration_seconds_blocking_scan: HistogramVec,
+}
+
+impl PrometheusMetrics {
+    /// new with prometheus register.
+    pub fn new(registry: Registry) -> Self {
+        // metadata
```

Comment:
We can remove this. Track metrics for metadata doesn't make sense :rofl:

Reply:
Question: is there meta cache in opendal? If yes, metadata latency is meaningful. 🤔

Reply:
Oh, `metadata` in `Accessor` just returns static info like `scheme`, `bucket_name` and so on. No real IO or syscall happened here.

Reply:
I see. Thanks, fixed.